packages/ember-core/src/components/form/file-upload.gts

```diff
@@ -0,0 +1,269 @@
+    isInvalid?: boolean;
+    isWarning?: boolean;
+    validatorKey?: string;
+    onAdd?: (files: File[]) => unknown;
+    onRemove?: (file: File) => unknown;
+  };
+  Blocks: {
+    default: [];
+  };
+  Element: HTMLDivElement;
+}
+
+class SelectedFileList extends Component<SelectedFileListSignature> {
+  @action
+  removeFile(index: number) {
+    this.args.onRemove?.(index);
+  }
+
+  <template>
+    {{#if @files}}
+      <div class="m-0 p-0 col-12">
+        <ul class={{classes "list-group list-group-flush rounded p-0 col-12 form-control" (if @isInvalid "is-invalid") (if @isWarning "is-warning")}}>
+          {{#each @files as |file index|}}
+            <li class="col-12 list-group-item d-flex flex-row align-items-center justify-content-between">
+              {{file.name}}
+              <Button data-test-remove class="btn-link" @onClick={{fn this.removeFile index}}>{{t "nrg.file-upload.remove"}}</Button>
+            </li>
+          {{/each}}
+        </ul>
+      </div>
+    {{else}}
+      <div class={{classes "form-control" (if @isInvalid "is-invalid") (if @isWarning "is-warning")}}>
+        <p class="text-muted m-0">{{t "nrg.file-upload.noFiles"}}</p>
+      </div>
+    {{/if}}
+  </template>
+}
+
+export default class FileUpload extends BoundValue<FileUploadSignature, File[]> {
+  validatorId?: string;
+
+  @tracked
+  inputElement?: HTMLInputElement;
+
+  @tracked
+  isDraggingOver = false;
+
+
+  constructor(owner: unknown, args: FileUploadSignature['Args']) {
+    super(owner, args);
+    registerDestructor(this, () => {
+      this.setupValidator();
+    });
+  }
+
+  get accept() {
+    return (this.args.accept as string[])?.join(', ') ?? '';
+  }
+
+  get dropzoneStyling() {
+    let style = "border-style: dashed !important;";
+    if (this.args.disabled) {
+      style += "cursor: not-allowed !important;";
+    }
+    return style;
+  }
+
+  filterDuplicateFiles(files: FileList): File[] {
+    if (!this.value || this.value!.length === 0) {
+      return Array.from(files);
+    }
+    const filteredFiles = [];
+    for (const newFile of files) {
+      if (!this.value!.some(file => file.name === newFile.name)) {
+        filteredFiles.push(newFile);
+      }
+    }
+    return filteredFiles;
+  }
+
+  @action
+  handleCancel(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+  }
+
+  @action
+  handleDragover(event: DragEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (this.args.disabled) {
+      return;
+    }
+    this.isDraggingOver = true;
+  }
+
+  @action
+  handleDrop(event: DragEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (this.args.disabled) {
+      return;
+    }
+    this.isDraggingOver = false;
+
+    const files = event.dataTransfer?.files;
+    this.change(files!)
+  }
+
+  @action
+  initInput(element: HTMLElement) {
+    if (element instanceof HTMLInputElement) {
+      this.inputElement = element;
+    }
+  }
```

Comment:
Since the `onInsert` modifier is attached to the `input` element, you're guaranteed this will be of type `HTMLInputElement`. I would try either changing the type of `element` to `HTMLInputElement` or cast it in the assignment:
```suggestion
  @action
  initInput(element: HTMLInputElement) {
    this.inputElement = element;
  }
```

Or this

```suggestion
  @action
  initInput(element: HTMLElement) {
    this.inputElement = element as HTMLInputElement;
  }
```